examples/threading/threading.c

```diff
@@ -28,6 +58,66 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      *
      * See implementation details in threading.h file comment block
      */
+
+    /*Allocating memory for thread data*/
+
+    printf("Dhiraj Bennadi*****************************\n");
+
+    struct thread_data *thread_data_ptr = (struct thread_data *)malloc(sizeof(struct thread_data));
+    if(thread_data_ptr == NULL)
+    {
+        ERROR_LOG("Dynamic Allocation Failed\n");
+        return false;
+    }
+
+
+    thread_data_ptr->mutexParameter = mutex;
+    thread_data_ptr->wait_to_obtain_ms = wait_to_obtain_ms;
+    thread_data_ptr->wait_to_release_ms = wait_to_release_ms;
+    thread_data_ptr->thread_complete_success = false;
+
+    
+
+    // /*Creation and Initialization of Mutex*/
+    // pthread_mutex_t m;
+
+    // if(pthread_mutex_init(&m, NULL) != 0)
+    // {
+    //     printf("Mutex Initialization Failed\n");
+    //     return false;
+
+    // }
+
+    /*Thread Creation*/
+    //pthread_t userThread;
+
+    retVal = pthread_create(thread, NULL, threadfunc, (void*)thread_data_ptr);
+
+    if(retVal == SUCCESS)
+    {
+        DEBUG_LOG("Thread ID = %ld************\n", *thread);
+        thread_data_ptr->threadId = *thread;
+        return true;
+
+    }
+    else
+    {
+        ERROR_LOG("Thread Creation Failed\n");
+    }
+
+    /*TODO: Tried this logic,but the threading is not joinable*/
+    if(thread_data_ptr != NULL)
+    {
+        free(thread_data_ptr);
+    }
+
+    //pthread_join(*thread , NULL);
+
+    if(retVal == SUCCESS)
+    {
+        return true;
```

Comment:
@dhirajbennadi I think the logic is correct here but it's confusing with the inline return above.  I think this is dead code and will never be executed.  You need to make sure you only `free(thread_data_ptr)` in failure cases.  It would probably be better to avoid the return at line 100 and put cleanup in the `else` block below